crypto/store/store_lib.c

```diff
@@ -422,7 +448,7 @@ EVP_PKEY *OSSL_STORE_INFO_get1_PKEY(const OSSL_STORE_INFO *info)
         return info->_.pkey;
     }
     OSSL_STOREerr(OSSL_STORE_F_OSSL_STORE_INFO_GET1_PKEY,
-                  OSSL_STORE_R_NOT_A_KEY);
+                  OSSL_STORE_R_NOT_A_PRIVATE_KEY);
```

Comment:
Didnt the code above say this could be a public key also? So I think the old error is better?

Reply:
The loader knows how to distinguish a public key (has public parts but no private parts) and a private key (also has private parts), so while *the rest of OpenSSL* has no way of distinguishing those, the application may still need to know, so OSSL_STORE now tells them apart, just like it has told apart an EVP_PKEY with just parameters.  See the message in `OSSL_STORE_INFO_get1_PARAMS`...

Reply:
Ah, you're referring to the comment in `load_key_cert_crl()` (`apps/lib/apps.c`), aren't you?  Maybe I need to elaborate on that one.

Reply:
I've rewritten the comment.